### PR TITLE
New wizard spell #2: Pacify

### PR DESCRIPTION
**Pull request by ghost**
Adds another new wizard spell: pacify. The spell creates an effect around the targeted area which pacifies any mobs by giving them small amounts of chillwax and oxycodone. The caster also receives the same amount of chillwax to discourage using the spell for braindead murderboning.

Cooldown is 45 seconds, can be lowered to 15 with two tiers of 15 second reduction each. Initial splash range is 3x3, can be upgraded to 5x5. Two power upgrades to increase the amount of reagents injected.

This one is a little less straightforward than the heal spell, so I'm willing to take any feedback on how it can be improved.

:cl:
 * rscadd: New pacify wizard spell.

### CODE REVIEW

**Comment by Irradiation** [comment]
Out of inane curiosity, can you still cast stuff like Fireball and all the other lethal spells while Chillwaxed?

**Comment by ghost** [comment]
No (mostly). Projectile spells all inherit similar behavior where they cancel on cast if the user is pacified. Most (but not all) non-projectile offensive spells have their own checks built in. I think all non-bundle offensive spells have checks, at least.

I initially implemented a pacify check that supersedes the existing ones for spells so that it wouldn't waste charge, but I didn't like how it turned out. I may revisit that later to make sure nobody comes up with workarounds.

**Comment by Irradiation** [comment]
So I take it that you can't cast Firebreath, Ring of Fire, Flesh to Stone and etc while chillwaxed?

**Comment by ghost** [comment]
Not firebreath or ring of fire, flesh to stone yes. I guess I'll just re-implement my check since the existing ones are so inconsistent.

**Comment by Killette2** [comment]
Not like flesh to stone is useful with its cooldown going off anyways if your target moves while you're casting it.

**Comment by ghost** [comment]
I re-implemented my flag that will cause spells to cancel when clicked so that you don't waste charge if you try to cast them while pacified. I'll have to tear up the existing snowflake checks later but they're not a priority since they'll never be tripped with this flag in usage.

If I missed a spell or unfairly tagged one as harmful please let me know so I can fix it.

File: code/modules/spells/targeted/pacify.dm [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	spell_flags = WAIT_FOR_CLICK
+	invocation = "YUKKRI SHEETI NAY"
+	invocation_type = SpI_SHOUT
+	range = 6
+
+	max_targets = 1
+
+	hud_state = "wiz_pacify"
+	var/power = 1
```
Comment (ShiftyRail) [inline code/modules/spells/targeted/pacify.dm]: Why not using `spell_levels[Sp_POWER]` directly instead

Comment (ghost) [reply code/modules/spells/targeted/pacify.dm]: Fixed

**Comment by Durgen** [comment]
3x3 and 5x5 seem kind of small

**Comment by ghost** [comment]
Disorient is similar and only has one target, I went off of that and made this AoE to counter the disadvantages. I can increase it to 5x5 and 7x7 if it turns out not to be very useful in its current state.

**Comment by tuypo1** [comment]
Not a fan of the existing pacifying stuff already, although I suppose it's slightly more bearable as a wizard spell instead of the other bullshit

**Comment by SonixApache** [comment]
be sure to add a spellbook

File: code/modules/spells/targeted/pacify.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	max_targets = 1
+
+	hud_state = "wiz_pacify"
+
+/spell/targeted/pacify/cast(list/targets, mob/user)
+	..()
+	if(targets)
+		if(user.reagents)
+			user.reagents.add_reagent(CHILLWAX, 4 + 4 * (spell_levels[Sp_POWER]/2))
```
Comment (Kurfursten) [inline code/modules/spells/targeted/pacify.dm]: 4 + 4 * POWER / 2 ?

Simplify this to just be 4 + POWER * 2

Comment (ghost) [reply code/modules/spells/targeted/pacify.dm]: Whoops, fixed.

**Pull request merged**